exammode/views.py

```diff
@@ -0,0 +1,56 @@
+from django.shortcuts import render, redirect
+from django.views import View, generic
+from django.http import HttpResponse
+from django.utils import timezone
+
+from userprofile.viewbase import UserProfileView
+from course.viewbase import EnrollableViewMixin
+from authorization.permissions import ACCESS
+
+from lib.viewbase import BaseTemplateView
+
+from .models import ExamSession, ExamAttempt
+
+# Create your views here.
+
+
+class ExamStartView(BaseTemplateView):
+    access_mode = ACCESS.STUDENT
+    template_name = "exammode/exam_start.html"
+
+    def get_context_data(self, **kwargs):
+        # Call the base implementation first to get the context
+        context = super(ExamStartView, self).get_context_data(**kwargs)
+        # Create any data and add it to the context
+        context['active_exams'] = ExamSession.active_exams.is_active()
+        return context
+
+
+class ExamDetailView(generic.DetailView):
+    model = ExamSession
+
+    def post(self, request, *args, **kwargs):
+        session = self.get_object()
```

Comment:
Even though this is a very simple `post` method, it would be still nice to check that the user actually pressed the button. (I assume that the user presses a button in a form before this method is called.) You can check the request POST data to see that the submit button was pressed. The button in the HTML form must have a `name` attribute so that it is included in the POST data. Is there a cancel button in the form and does it redirect the user somewhere else than this method?

Reply:
There is link to the previous page in the html template.

Reply:
A link to the previous page does basically the same thing as a cancel button should. If the form can only be posted in one way, then it is not necessary to check if the POST data contains the value of the submit button.

Reply:
I added the check anyway. It can be removed if it useless and only makes code harder to read.